pages/courses/[id].js

```diff
@@ -438,8 +438,8 @@ function Course({ course, currentDate }) {
   )
 }
 
-export async function getServerSideProps({ params }) {
-  const course = await getCourse(params.id)
+export async function getServerSideProps({ params, locale }) {
+  const course = await getCourse(params.id, locale)
```

Comment:
again, locale should be something decided only on view, not on server side props definition.